client.go

```diff
@@ -304,9 +315,64 @@ func (client *client) GetOffset(topic string, partitionID int32, time int64) (in
 	return offset, err
 }
 
+func (client *client) Coordinator(consumerGroup string) (*Broker, error) {
+	client.lock.RLock()
+	coordinatorID, coordinatorCached := client.coordinators[consumerGroup]
+	client.lock.RUnlock()
+
+	if !coordinatorCached {
+		var err error
+		err = client.RefreshCoordinator(consumerGroup)
+		if err != nil {
```

Comment:
the above three lines can be one `if err := client.RefreshCoordinator(consumerGroup); err != nil {`